src/Pecee/Http/Input/InputHandler.php

```diff
@@ -201,11 +230,12 @@ protected function parseInputItem(array $array): array
         foreach ($array as $key => $value) {
 
             // Handle array input
-            if (\is_array($value) === true) {
-                $value = $this->parseInputItem($value);
+            if (\is_array($value) === false) {
```

Comment:
I will look over it but I used the same code in my current project and it worked.

Reply:
I need to create some unit tests for when items are nested arrays, because I got some funky results with the current release ($_POST['myData'] is an array in the above example and i think the `InputHandler` in your code will treat all sub-items as `InputItems` which they aren't).

I could be wrong, that's why the unit-tests makes life easier even though it can be a bit tedious to create.

Reply:
Ahh, I know what you mean. Yes the sub elements of arrays are threaded as Input Items. It was intentionally but I will change that.

Reply:
It's the same behavior in the current release. It's not that common to use nested arrays like that, but when you do you get some funky results where indexes are missing etc because everything is wrapped as a `InputItem`. I changed it in `v4-development`.

I'll add some more unit-tests for the `InputHandler` soon, that makes it a little easier when making new features.

Reply:
👍🏻

Reply:
You changed the function to the following in v4-development:
 ```
protected function parseInputItem(array $array): array
{
    $list = [];

    foreach ($array as $key => $value) {

        // Handle array input
        if (\is_array($value) === true) {
            $value = $this->parseInputItem($value);
        }

        $list[$key] = new InputItem($key, $value);
    }

    return $list;
}
```
But this will thread sub arrays as InputItems...
if the value is an array it will be changed into an array of InputItems, because of the rekursion and then set as the value of the InputItem.

When you don't want to have array contents wrapped into an InputItem you have to do the following:
```
protected function parseInputItem(array $array): array
{
    $list = [];

    foreach ($array as $key => $value) {
        $list[$key] = new InputItem($key, $value);
    }

    return $list;
}
```
We have mixed InputItem values now. So we can the an array as value.
Do I misunderstand something?

~ Marius

Reply:
Or do you want to handle array contents also as InputItems? Because if the opposite is the case you do not need the `getValueFromArray` function^^

Reply:
I added better support nested array (before it would sometimes overwrite the index name on certain nested arrays), but more importantly I fixed `InputHandler::value` method so it ALWAYS returns the original value and not the `InputItem` (that's what getValueFromArray is for). Before it would return an array of InputItems, which is not the original value. 

Maybe it should be renamed to `parseValueFromArray` maybe that would make more sense?

Reply:
Yes, this would sound better I think.
Ahh, now I got your point 👍🏻

Reply:
What is if you run `post()` get an `InputItem` and call `getValue()`? Shouldn't we add `parseValueFromArray` to the getValue() function? Wouldn't it be useless then?

Reply:
You have a good point. You're right - it would make sense 👍

Reply:
So why should we still parse input array contents?

Reply:
Ahh, no. Let's still parse it, but on the InputItem you can now call `getValue` (get the normal contents) or `getInputItem()` to get the InputItems if it is an array.

Reply:
Some values are nested like arrays if people use things like `input name="test[]` or `input name="test[data][]`. It's a little confusion, maybe the right thing to do would be to add some complex unit-tests and see what makes sense. Because some of the things I'm not entirely sure about, but I know it was added for a reason.

I'm planning on adding input-validation support too in the next version.

Reply:
You always think about form contents but what about ajax contents? I submit data via ajax with nested arrays. You don't do that with x-urlenc or form-data but with json content-type.

Can I just add the input-validation to the new inputhandler? I just started with that

Reply:
That's a great idea! That gives the control back to the developer, he can chose what output he wants.

Reply:
Okay 👍🏻